flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamConfigChainer.java

```diff
@@ -262,4 +289,143 @@ public StreamConfigChainer<OWNER> name(String name) {
     public void setBufferTimeout(int bufferTimeout) {
         this.bufferTimeout = bufferTimeout;
     }
+
+    /** Helper class to build operator node. */
+    public static class StreamConfigEdgeChainer<OWNER, IN, OUT> {
```

Comment:
nit: for the future, it would be easier to review, if you added this builder class in a separate commit, preceding your changes. Then in your bug fix you could have only added `outEdgesInOrder` and `setTailNonChainedOutputs ` fields to the builder, which would make the bug fix commit smaller and easier to review.